services/bitshares_elasticsearch_client.py

```diff
@@ -28,7 +28,7 @@ def get_markets(self, from_date, to_date, base=None, quote=None):
             "query": {
                 "bool": {
                     "filter": [
-                        { "term": { "operation_type": 4 } },
+                        { "term": { "operation_type": 4 } }, # NOTE: may logically return duplicate data since not filtering by `is_maker == true`
```

Comment:
The duplication is logical, because on every order match, two `fill_order_operation` entries are stored, one for each side, and their `fill_price` is exactly the same. So the `fill_price__quote__asset_id` filter does not work as intended. The `receives__asset_id` filter (my code) may work.

We should not filter by `is_maker` here though. Actually we need the duplicate data. I'll update the comment.

Reply:
Ah yea I remember... The explorer API that I am using for blocksights is a complete rewrite

```
def get_markets(from_date, to_date, base=None, quote=None):
    ...
    s = s.extra(size=0)
    s = s.source([])
    s = s.filter(
        Q('term', operation_type=4) &
        Q('range', block_data__block_time={'gte': from_date, 'lte': to_date}) &
        Q('term', operation_history__op_object__is_maker=False)
    )
    if base:
        s = s.filter(
            Q('term', operation_history__op_object__pays__asset_id=base)
        )
    if quote:
        s = s.filter(
            Q('term', operation_history__op_object__receives__asset_id=quote)
        )
    a = A(
        "composite",
        sources=[
            { "base_asset_id": { "terms" : { "field": "operation_history.op_object.pays.asset_id.keyword" } } },
            { "quote_asset_id": { "terms" : { "field": "operation_history.op_object.receives.asset_id.keyword" } } }
        ],
        size=10000
    ).metric(
        'base_volume', 'sum', field='operation_history.op_object.pays.amount'
    ).metric(
        'quote_volume', 'sum', field='operation_history.op_object.receives.amount'
    )
    s.aggs.bucket(
        'pairs',
        a
    )
    response = s.execute()
```

Reply:
It's correct then.
So I think the issue on blocksights is that we should not add amounts in different assets together.